src/routes/services/passport-strategies.js

```diff
@@ -0,0 +1,40 @@
+import passport from 'passport';
+import dotenv from 'dotenv';
+import models from '../../db/models';
+
+dotenv.config();
+
+const { User } = models;
+const FacebookStrategy = require('passport-facebook').Strategy;
+
+const credentials = {
+  facebook: {
+    clientID: process.env.FACEBOOK_APP_CLIENT_ID,
+    clientSecret: process.env.FACEBOOK_APP_SECRET,
+    callbackURL: 'http://localhost:3000/api/v1/auth/facebook/callback',
+    profileFields: ['id', 'email', 'name'],
+  }
+};
+
+const facebookAuth = async (accessToken, refreshToken, profile, done) => {
+  try {
+    const currentUser = await User.findOrCreate({
+      where: { socialId: profile.id },
+      defaults: {
+        firstName: profile.name.givenName,
+        lastName: profile.name.familyName,
+        username: profile.emails[0].value,
```

Comment:
Why are we giving username the value of email?

Reply:
The username of the facebook user might not be unique on our application. The default username is now set to be the email, since emails are unique.